frontend/benefit/handler/src/components/applicationsArchive/ApplicationArchiveList.tsx

```diff
@@ -0,0 +1,167 @@
+  const theme = useTheme();
+
+  interface TableTransforms {
+    id?: string;
+    companyName?: string;
+    status?: APPLICATION_STATUSES;
+    endDate?: string;
+  }
+
+  const getTransformForArchivedStatus = ({
+    status,
+  }: TableTransforms): JSX.Element => (
+    <$TagWrapper $colors={getTagStyleForStatus(status)}>
+      <Tag>
+        {t(
+          `common:applications.list.columns.applicationStatuses.${String(
+            status
+          )}`
+        )}
+      </Tag>
+    </$TagWrapper>
+  );
+
+  const cols = [
+    {
+      transform: ({ id, companyName }: TableTransforms) => (
+        <$Link href={`/application?id=${String(id)}`}>
+          {String(companyName)}
+        </$Link>
+      ),
+      headerName: getHeader('companyName'),
+      key: 'companyName',
+      isSortable: true,
+    },
+    {
+      headerName: getHeader('companyId'),
+      key: 'companyId',
+      isSortable: true,
+    },
+    {
+      headerName: getHeader('applicationNum'),
+      key: 'applicationNum',
+      isSortable: true,
+    },
+    {
+      headerName: getHeader('employeeNameArchive'),
+      key: 'employeeName',
+      isSortable: true,
+    },
+    {
+      headerName: getHeader('handledAt'),
+      key: 'handledAt',
+      isSortable: true,
+      customSortCompareFunction: sortFinnishDate,
+    },
+    {
+      headerName: getHeader('calculationEndDate'),
+      key: 'calculationEndDate',
+      isSortable: true,
+      customSortCompareFunction: sortFinnishDate,
+    },
+    {
+      transform: getTransformForArchivedStatus,
+      headerName: t(`${translationsBase}.columns.statusArchive`)?.toString(),
+      key: 'status',
+      isSortable: true,
+    },
+  ];
+
+  const hasSearchLoadedWithResults =
+    !isSearchLoading &&
+    submittedSearchString !== '' &&
+    searchString.length > 0 &&
+    data?.length > 0;
+
+  const hasSearchLoadedWithNoResults =
+    !isSearchLoading &&
+    submittedSearchString !== '' &&
+    searchString !== '' &&
+    data?.length === 0;
+
+  const hasCurrentlyNoSearch =
+    !isSearchLoading &&
+    list.length > 0 &&
+    (submittedSearchString === '' || searchString === '');
+
+  return (
+    <>
+      {hasCurrentlyNoSearch && (
+        <>
+          <$ArchiveCount>{`${t(`${translationsBase}.total.count`, {
+            count: list?.length || 0,
+          })}`}</$ArchiveCount>
+          <Table
+            indexKey="id"
+            theme={theme.components.table}
+            rows={list || []}
+            cols={cols}
+          />
+        </>
+      )}
+
+      {isSearchLoading && (
+        <>
+          <$ArchiveCount>{t('common:search.performingSearch')}</$ArchiveCount>
+          <LoadingSkeleton
+            borderRadius={0}
+            baseColor={theme.colors.fog}
+            height={50}
+          />
+          <LoadingSkeleton height={67} />
+        </>
+      )}
+
+      {hasSearchLoadedWithResults && (
```

Comment:
I wonder if the following situation could be made a bit less confusing:
1) Initiate any search, press enter (search results appear)
2) Clear the search (full list appears)
3) Start typing (previous results appear again until a new search is submitted)

Functionally that's not really different from modifying the existing search, but moving from the cleared list back to unrelated results might feel more jarring. The results could maybe be cleared when the search text is cleared.